base/range.jl

```diff
@@ -415,6 +415,13 @@ end
 
 function getindex{T}(v::Range{T}, i::Integer)
     @_inline_meta
+    ret = convert(T, first(v) + i - 1)
+    @boundscheck ((i > 0) & (ret <= v.stop) & (ret >= v.start)) || throw_boundserror(v, i)
+    ret
+end
+
+function getindex{T}(v::Range{T}, i::Integer)
```

Comment:
Actually a good outcome: you pushed me to check general `Range` rather than being lazy and just fixing `UnitRange` (as was my intention)...and got a 5x improvement (!!) out of it.

While it seems nice, it makes me nervous I'm missing something important. Anyone who knows the range code well is invited to chime in.

I'll also run the benchmarks once https://github.com/JuliaCI/BaseBenchmarks.jl/pull/8 merges.